pkg/kubernetes/client.go

```diff
@@ -15,3 +24,47 @@ func GetDiscoveryClient(mgr manager.Manager) (*discovery.DiscoveryClient, error)
 	}
 	return dc, nil
 }
+
+// DeleteAndWait deletes the given runtime object and waits until it is fully deleted
+func DeleteAndWait(c client.Client, obj runtime.Object, options ...client.DeleteOption) error {
+	err := c.Delete(context.TODO(), obj, options...)
+
+	if err != nil {
+		key := ObjectKey(obj)
+
+		if kerrors.IsNotFound(err) {
+			// Obj is already deleted, we can return directly
+			clog.V(6).Printf("Deleting obj %s/%s is already NotFound, return now", key.Namespace, key.Name)
+			return nil
+		}
+		return fmt.Errorf("failed to delete %s/%s: %v", key.Namespace, key.Name, err)
+	}
+
+	return WaitForDelete(c, obj)
+}
+
+// WaitForDelete waits for the provided runtime object to be deleted from cluster
+func WaitForDelete(c client.Client, obj runtime.Object) error {
+	key := ObjectKey(obj)
+	clog.V(6).Printf("Waiting for obj %s/%s to be finally deleted", key.Namespace, key.Name)
+
+	// Wait for resources to be deleted.
+	return wait.PollImmediate(250*time.Millisecond, 30*time.Second, func() (done bool, err error) {
+		err = c.Get(context.TODO(), key, obj.DeepCopyObject())
+		clog.V(8).Printf("Fetched %s/%s to wait for delete: %v", key.Namespace, key.Name, err)
```

Comment:
questioning this verbose levels...   level 7-9 tend to be deconstruction of serialized objects or deep http calls.   These seem like level 2 and 3 type stuff (obviously subjective)

Reply:
I've adjusted them a bit to 6 - This one especially is in a tight loop and can get quite spammy that's why I didn't want to have it in the higher ones